File: 1/1.lua
---------------
-- Variables --
---------------

-- Input the file into a table
local file = io.open("input.txt", "r");
local arr = {}
for line in file:lines() do
   table.insert (arr, line);
end

local sMilli = os.clock()

---------------
-- Functions --
---------------

function summed2020(x,y)
    if x + y == 2020 then
        return true
    end
    return false
end

for i, x in pairs(arr) do
    x = tonumber(x)

    for i, y in pairs(arr) do
        y = tonumber(y)
        if summed2020(x,y) == true then
            local eMilli = os.clock()
            print("The answer is: "..x*y..". Duration: "..eMilli - sMilli)
            os.exit()
        end
    end
end

File: 1/1.rb
#!/usr/bin/env ruby

# Variables
input = IO.readlines("input.txt")

# Functions
def summed2020?(x,y)
    return true if x + y == 2020
    return false
end


input.each do |e|
    e = e.to_i
    input.each do |x|
        x = x.to_i
        if summed2020?(e,x)
            printf "The answer: #{e * x}\n"
            exit
        end
    end
end
